networkx/algorithms/tests/test_richcore.py

```diff
@@ -0,0 +1,16 @@
+import networkx as nx
+from nose.tools import *
+
+def test_richcore():
+    G=nx.Graph()
+    G.add_edge('a','b',weight=0.6)
+    G.add_edge('a','c',weight=0.2)
+    G.add_edge('c','d',weight=0.1)
+    G.add_edge('c','e',weight=0.7)
+    G.add_edge('c','f',weight=0.9)
+    G.add_edge('a','d',weight=0.3)
+
+    sigmas, ranked_nodes, r_star = nx.richcore.extract_rich_core(G, weight='weight')
+    assert_equal(sigmas, [0, 19.0, 19.0, 19.0, 10.999999999999998, 30.0])
```

Comment:
tests with floating point numbers are tricky to get exactly equal. 
You might consider ```from nose.utils import assert_almost_equal``` for the tests.

There is also a nice way to make sure division works for floating point for python2 and 3. Use this at the top of the rich_core.py module:  ```from __future__ import division```     Then you don't need to use ```1.*``` when you divide.